components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/ApplicationKeyMappingUtil.java

```diff
@@ -0,0 +1,104 @@
+import org.wso2.carbon.apimgt.rest.api.store.v1.dto.ApplicationKeyGenerateRequestDTO;
+import org.wso2.carbon.apimgt.rest.api.store.v1.dto.ApplicationTokenDTO;
+import org.wso2.carbon.apimgt.rest.api.util.exception.InternalServerErrorException;
+
+import java.util.Arrays;
+import java.util.Map;
+
+public class ApplicationKeyMappingUtil {
+
+    private static final Log log = LogFactory.getLog(ApplicationKeyMappingUtil.class);
+
+    @SuppressWarnings("unchecked")
+    public static ApplicationKeyDTO fromApplicationKeyToDTO(Map<String, Object> keyDetails, String applicationKeyType) {
+        ApplicationKeyDTO applicationKeyDTO = new ApplicationKeyDTO();
+        applicationKeyDTO.setConsumerKey((String) keyDetails.get(APIConstants.FrontEndParameterNames.CONSUMER_KEY));
+        applicationKeyDTO
+                .setConsumerSecret((String) keyDetails.get(APIConstants.FrontEndParameterNames.CONSUMER_SECRET));
+        applicationKeyDTO.setKeyState((String) keyDetails.get(APIConstants.FrontEndParameterNames.KEY_STATE));
+        try {
+            String appDetailsString = (String) keyDetails.get(ApplicationConstants.OAUTH_APP_DETAILS);
+            if (appDetailsString != null) {
+                JSONObject appDetailsJsonObj = (JSONObject) new JSONParser().parse(appDetailsString);
+                if (appDetailsJsonObj != null) {
+                    applicationKeyDTO.setKeyType(ApplicationKeyDTO.KeyTypeEnum.valueOf(applicationKeyType));
+                    String supportedGrantTypes = (String) appDetailsJsonObj
+                            .get(ApplicationConstants.OAUTH_CLIENT_GRANT);
+                    if (supportedGrantTypes != null) {
+                        applicationKeyDTO.setSupportedGrantTypes(Arrays.asList(supportedGrantTypes.split(" ")));
+                    }
+                    String callbackUrl = (String) appDetailsJsonObj.get(ApplicationConstants.OAUTH_REDIRECT_URIS);
+                    applicationKeyDTO.setCallbackUrl(callbackUrl);
+                }
+            }
+
+            ApplicationTokenDTO tokenDTO = new ApplicationTokenDTO();
+            tokenDTO.setValidityTime((Long)keyDetails.get(APIConstants.AccessTokenConstants.VALIDITY_TIME));
+            tokenDTO.setAccessToken((String)keyDetails.get(APIConstants.AccessTokenConstants.ACCESS_TOKEN));
+            String[] tokenScopes = (String[])keyDetails.get(APIConstants.AccessTokenConstants.TOKEN_SCOPES);
+            if (tokenScopes != null) {
+                tokenDTO.setTokenScopes(Arrays.asList(tokenScopes));
+            }
+
+            applicationKeyDTO.setToken(tokenDTO);
+        } catch (ParseException e) {
+            String errorMsg = "Error while parsing application details string";
+            log.error(errorMsg, e);
+            throw new InternalServerErrorException(errorMsg, e);
+        }
+        return applicationKeyDTO;
+    }
+
+    public static ApplicationKeyDTO fromApplicationKeyToDTO(APIKey apiKey) {
```

Comment:
Do we need this method? seems like it is not used

Reply:
It will be used in other rest api implementations (according to old rest api) so we better leave it here for now